Tests/FluentAssertions.Specs/Configuration/FormattingOptionsSpecs.cs

```diff
@@ -0,0 +1,216 @@
+    [Fact]
+    public void When_global_formatting_settings_are_modified()
+    {
+        AssertionConfiguration.Current.Formatting.UseLineBreaks = true;
+        AssertionConfiguration.Current.Formatting.MaxDepth = 123;
+        AssertionConfiguration.Current.Formatting.MaxLines = 33;
+
+        AssertionScope.Current.FormattingOptions.UseLineBreaks.Should().BeTrue();
+        AssertionScope.Current.FormattingOptions.MaxDepth.Should().Be(123);
+        AssertionScope.Current.FormattingOptions.MaxLines.Should().Be(33);
+    }
+
+    [Fact]
+    public void When_a_custom_formatter_exists_in_any_loaded_assembly_it_should_override_the_default_formatters()
+    {
+        // Arrange
+        AssertionConfiguration.Current.Formatting.ValueFormatterDetectionMode = ValueFormatterDetectionMode.Scan;
+
+        var subject = new SomeClassWithCustomFormatter
+        {
+            Property = "SomeValue"
+        };
+
+        // Act
+        string result = Formatter.ToString(subject);
+
+        // Assert
+        result.Should().Be("Property = SomeValue", "it should use my custom formatter");
+    }
+
+    [Fact]
+    public void When_a_base_class_has_a_custom_formatter_it_should_override_the_default_formatters()
+    {
+        // Arrange
+        AssertionConfiguration.Current.Formatting.ValueFormatterDetectionMode = ValueFormatterDetectionMode.Scan;
+
+        var subject = new SomeClassInheritedFromClassWithCustomFormatterLvl1
+        {
+            Property = "SomeValue"
+        };
+
+        // Act
+        string result = Formatter.ToString(subject);
+
+        // Assert
+        result.Should().Be("Property = SomeValue", "it should use my custom formatter");
+    }
+
+    [Fact]
+    public void When_there_are_multiple_custom_formatters_it_should_select_a_more_specific_one()
+    {
+        // Arrange
+        AssertionConfiguration.Current.Formatting.ValueFormatterDetectionMode = ValueFormatterDetectionMode.Scan;
+
+        var subject = new SomeClassInheritedFromClassWithCustomFormatterLvl2
+        {
+            Property = "SomeValue"
+        };
+
+        // Act
+        string result = Formatter.ToString(subject);
+
+        // Assert
+        result.Should().Be("Property is SomeValue", "it should use my custom formatter");
+    }
+
+    [Fact]
+    public void When_a_base_class_has_multiple_custom_formatters_it_should_work_the_same_as_for_the_base_class()
+    {
+        // Arrange
+        AssertionConfiguration.Current.Formatting.ValueFormatterDetectionMode = ValueFormatterDetectionMode.Scan;
+
+        var subject = new SomeClassInheritedFromClassWithCustomFormatterLvl3
+        {
+            Property = "SomeValue"
+        };
+
+        // Act
+        string result = Formatter.ToString(subject);
+
+        // Assert
+        result.Should().Be("Property is SomeValue", "it should use my custom formatter");
+    }
+
+    [Fact]
+    public void When_no_custom_formatter_exists_in_the_specified_assembly_it_should_use_the_default()
```

Comment:
It seems some behavior changed.
This test used to cover the `return null` in `AttributeBasedFormatter.GetFormatter`

Reply:
I'm not sure what you mean. This is what it looked like before my merge. https://github.com/fluentassertions/fluentassertions/blob/3d9334af01c211749f8113341646731be0b544e3/Tests/FluentAssertions.Specs/Formatting/FormatterSpecs.cs#L1022

Reply:
[Before](https://coveralls.io/builds/71581506/source?filename=Src%2FFluentAssertions%2FFormatting%2FAttributeBasedFormatter.cs#L59):
![image](https://github.com/user-attachments/assets/b1446a24-e7ec-4cb5-8d41-aeb250357020)
[After](https://coveralls.io/builds/71582076/source?filename=Src%2FFluentAssertions%2FFormatting%2FAttributeBasedFormatter.cs#L57):
![image](https://github.com/user-attachments/assets/20410a27-5b97-48e0-a32c-96b496ca1c64)

Before this PR `When_no_custom_formatter_exists_in_the_specified_assembly_it_should_use_the_default` was one of the tests exercising that `return null`.